### PR TITLE
[JENKINS-35285] Upgrade to parent POM

### PR DESCRIPTION
**Pull request by kwhetstone**
Updated dependencies for maven-release, findbugs, and annotations.  Kept jenkins.version at 1.566.  Passed tests on that and 2.6.  Implements [JENKINS-35285](https://issues.jenkins-ci.org/browse/JENKINS-35285)

@reviewbybees

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -24,8 +24,7 @@
   </licenses>
   
   <properties>
-    <findbugs-maven-plugin.version>3.0.1</findbugs-maven-plugin.version>
-    <findbugs.failOnError>true</findbugs.failOnError>
+    <jenkins.version>1.566</jenkins.version>
   </properties>
```
Comment (jglick) [inline pom.xml]: :bug: forgot

```xml
<java.level>6</java.level>
```

Wish we could default it based on `jenkins.version`, but for now it has to be set manually.

Comment (kwhetstone) [inline pom.xml]: Added java.level

**Comment by jglick** [comment]
:bee:

**Comment by armfergom** [comment]
🐝

**Comment by kwhetstone** [comment]
@reviewbybees done

**Pull request merged**